core/trino-main/src/main/java/io/trino/Session.java

```diff
@@ -337,7 +337,7 @@ public Session beginTransactionId(TransactionId transactionId, TransactionManage
                 continue;
             }
             CatalogHandle catalogHandle = transactionManager.getCatalogHandle(transactionId, catalogName)
-                    .orElseThrow(() -> new TrinoException(NOT_FOUND, "Session property catalog does not exist: " + catalogName));
+                    .orElseThrow(() -> new TrinoException(CATALOG_NOT_FOUND, "Session property catalog '%s' not found".formatted(catalogName)));
```

Comment:
Sorry .. but what is that message supposed to mean? Is this about a catalog session property in a specific catalog? .. and what is not found .. the catalog or the catalog session property? 

I realize the old message makes no sense btw.. but the new one makes even less sense.

Reply:
I think the catalog is not found, but it's in the context of parsing session properties for a catalog. So the user has session properties set for a catalog, and the catalog was not found. I'm not entirely sure what a better message here would be. I think that `Catalog 'x' not found` would be clearer, but then you lose the context that it was not found while validating the session properties.

Reply:
Maybe `Catalog 'x' not found while validating session properties`? Or is that too verbose?

Reply:
Does this come up when the catalog is not found or when the catalog session property is invalid for that catalog? Or both? Depending on scenario we might need different message.

Reply:
When the catalog is not found

Reply:
Lets maybe go with 

Catalog x for session property y not found

Reply:
There isn't one particular session property, it groups them by catalog, so it's ALL of the session properties defined for a catalog. Maybe "Catalog 'x' in session properties not found"?